pages/submit.js

```diff
@@ -458,7 +458,21 @@ class SubmitDapp extends React.Component {
     values: {},
     loading: false,
     success: false,
-    errors: {}
+    errors: {},
+    referralCode: null,
+    refSource: null
+  }
+
+  componentDidMount() {
+    const { search } = document.location
+    if (search) {
+      const referralCode = search.match(/referralCode=(\w+)/)[1]
+      const refSource = search.match(/refSource=(\w+)/)[1]
+      this.setState({ // eslint-disable-line react/no-did-mount-set-state
```

Comment:
I think we could refine this a bit and only do `setState` if the `referralCode` || `refSource` are present. I think this would do setState even if they are not there.

Reply:
I will change this based on the other comment, but agreed.